src/main/java/com/flansmod/common/driveables/DriveableType.java

```diff
@@ -448,682 +474,800 @@ public void preRead(TypeFile file) {
+                    FlansMod.logPackError(file.name, packName, shortName, "Adding collision mesh failed", split, ex);
+                }
             }
 
-            // ICBM Mod Radar
-            else if (split[0].equals("OnRadar"))
-                onRadar = split[1].equals("True");
-
-            else if (split[0].equalsIgnoreCase("AddParticle") || split[0].equalsIgnoreCase("AddEmitter")) {
-                ParticleEmitter emitter = new ParticleEmitter();
-                emitter.effectType = split[1];
-                emitter.emitRate = Integer.parseInt(split[2]);
-                emitter.origin = new Vector3f(split[3], shortName);
-                emitter.extents = new Vector3f(split[4], shortName);
-                emitter.velocity = new Vector3f(split[5], shortName);
-                emitter.minThrottle = Float.parseFloat(split[6]);
-                emitter.maxThrottle = Float.parseFloat(split[7]);
-                emitter.minHealth = Float.parseFloat(split[8]);
-                emitter.maxHealth = Float.parseFloat(split[9]);
-                emitter.part = split[10];
-                //Scale from model coords to world coords
-                emitter.origin.scale(1.0f / 16.0f);
-                emitter.extents.scale(1.0f / 16.0f);
-                emitter.velocity.scale(1.0f / 16.0f);
-                emitters.add(emitter);
+
+            splits = ConfigUtils.getSplitsFromKey(config, new String[] { "LeftLinkPoint" });
+            for (String[] split : splits) {
+                try {
+                    leftTrackPoints.add(new Vector3f(split[1]));
+                } catch (Exception ex) {
+                    FlansMod.logPackError(file.name, packName, shortName, "Adding LeftLinkPoint failed", split, ex);
+                }
             }
-        } catch (Exception e) {
-            if (split != null) {
-                StringBuilder msg = new StringBuilder(" : ");
-                for (String s : split) msg.append(" ").append(s);
-                FlansMod.log("Errored reading " + file.name + msg);
-            } else {
-                FlansMod.log("Errored reading " + file.name);
+
+            splits = ConfigUtils.getSplitsFromKey(config, new String[] { "RightLinkPoint" });
+            for (String[] split : splits) {
+                try {
+                    rightTrackPoints.add(new Vector3f(split[1]));
+                } catch (Exception ex) {
+                    FlansMod.logPackError(file.name, packName, shortName, "Adding RightLinkPoint failed", split, ex);
+                }
             }
-            if (FlansMod.printStackTrace) {
-                e.printStackTrace();
+
+
+
+            trackLinkLength = ConfigUtils.configFloat(config, "TrackLinkLength", trackLinkLength);
+
+            // ICBM Mod Radar
+            onRadar = ConfigUtils.configBool(config, "OnRadar", onRadar);
+
+
+            splits = ConfigUtils.getSplitsFromKey(config, new String[] { "AddParticle", "AddEmitter" });
```

Comment:
Is this still case insensitive? The original code used `equalsIgnoreCase`.

Reply:
Yup - everything's stored/retrieved in lower case.
https://github.com/Unknown025/Flans-Mod-Plus/blob/wip_config_rewrite/src/main/java/com/flansmod/utils/ConfigUtils.java#L11

https://github.com/Unknown025/Flans-Mod-Plus/blob/wip_config_rewrite/src/main/java/com/flansmod/utils/ConfigMap.java#L21